### ISSUE TITLE
SOCIALACCOUNT_ENABLED is False when custom AppConfig is used

### ISSUE THREAD

**Issue by Flimm** [pennersr/django-allauth#2853]
I'm using Django 3.2. In order to configure allauth to use `AutoField` in Django 3.2, I created this custom AppConfig in `projectname/apps.py`:

```python
from allauth.account.apps import AccountConfig
from allauth.socialaccount.apps import SocialAccountConfig
from django.apps import AppConfig


class ModifiedAccountConfig(AccountConfig):
    default_auto_field = 'django.db.models.AutoField'

class ModifiedSocialAccountConfig(SocialAccountConfig):
    default_auto_field = 'django.db.models.AutoField'
```

Then in `settings.py`, I modified `INSTALLED_APPS` to replace `allauth.account` and `allauth.socialaccount` like this:

```python
INSTALLED_APPS = [
    # ....
    # remove: "allauth.account",
    "projectname.apps.ModifiedAccountConfig",
    # remove: "allauth.socialaccount",
    "projectname.apps.ModifiedSocialAccountConfig",
]
```

For reference, `urls.py` contained:

```python
from django.urls import include, path

urlpatterns = [
    path('accounts/', include('allauth.urls'),
]
```

This seemed to work as expected.

However, I had a Django template which included this code:

```
<a href="{% url "socialaccount_connections" %}">example</a>
```

When this template is rendered, it causes this error:

```
NoReverseMatch at /profile/

Reverse for 'socialaccount_connections' not found. 'socialaccount_connections' is not a valid view function or pattern name.
```

It turns out that this URL is added in `allauth/socialaccount/urls.py`. This file is in turn activated in `allauth/urls.py` in this code:

```python
if app_settings.SOCIALACCOUNT_ENABLED:
    urlpatterns += [path("social/", include("allauth.socialaccount.urls")]
```

How does `SOCIALACCOUNT_ENABLED` get set? The answer is in `allauth/app_settings.py`:

```python
from django.conf import settings

SOCIALACCOUNT_ENABLED = "allauth.socialaccount" in settings.INSTALLED_APPS
```

The bug in this line is that it will set `SOCIALACCOUNT_ENABLED` to False if it contains an `AppConfig` that merely inherits from `SocialAccountConfig`. Instead, it should know how to handle this case and set `SOCIALACCOUNT_ENABLED` to True.

**Comment by Flimm** [comment]
Here is the workaround that I ended up using. In `projectname/settings.py`, I added these lines at the end:

```python
# Monkey-patch fix:
import allauth.app_settings
allauth.app_settings.SOCIALACCOUNT_ENABLED = True
```

However, I'm not sure if this is a good solution, as I'm not sure how the circular dependency between `projectname.settings` and `allauth.app_settings` is resolved.

**Comment by Flimm** [comment]
The latest release on PyPI does contain the fix, now.

**Comment by derek-adair** [comment]
I could be wrong but wont just manually setting `SOCIAL_ACCOUNT_ENABLED=True` in your settings work?

**Comment by pennersr** [comment]
@derek-adair It is not a setting, it is a derived constant using:

```
SOCIALACCOUNT_ENABLED = "allauth.socialaccount" in settings.INSTALLED_APPS
```

It likely needs to be replaced with:

```
django_apps.is_installed("allauth.socialaccount")
```

So this is a valid issue.

**Comment by derek-adair** [comment]
Ahhh. i see... app_settings imports django.settings....  However, I don't think that will work.  The user is overriding the app label here so I believe the app label should also be `projectname.apps.ModifiedSocialAccountConfig`.  It seems this is simply used to set up the social urls.  

1. Is there a reason not to make this a configurable setting?
2. To guard against users setting this to True and not having it installed, throw a warning?
3. Is this something worth supporting?? it seems like it.  I've not spent really any time contemplating how to support this kind of thing on my own projects, so I am very out of my element on this one.

**Comment by pennersr** [comment]
It would work, as `is_installed()` checks against the config `name` property, which is not overridden by `ModifiedAccountConfig`.

**Comment by pennersr** [comment]
Fixed via 0dadd0d7

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/2853
